packet_queue/nfqueue.py

```diff
@@ -60,18 +61,33 @@ def configure(protocol, port, pipes, interface):
   reactor.addReader(reader)
 
 
-def add(protocol, port, interface):
+def add(protocol, port, interface, role, use_ipv4, use_ipv6):
+  """Adds iptables NFQUEUE rules: one each for INPUT/OUTPUT/IPv4/IPv6."""
+  # Add IPv4 handlers
+  if use_ipv4:
+    _add(protocol, port, interface, role, iptc.Table, iptc.Rule)
+  # Add IPv6 handlers
+  if use_ipv6:
+    _add(protocol, port, interface, role, iptc.Table6, iptc.Rule6)
+
+
+def _add(protocol, port, interface, role, table_cls, rule_cls):
   """Adds iptables NFQUEUE rules: one each for INPUT and OUTPUT."""
-  table = iptc.Table(iptc.Table.FILTER)
+  table = table_cls(iptc.Table.FILTER)
 
-  params =  [
-    ('INPUT', 'in_interface', 'dport', UP_QUEUE),
-    ('OUTPUT', 'out_interface', 'sport', DOWN_QUEUE),
+  if role == 'server':
+    ports = ('dport', 'sport')
+  else:
+    ports = ('sport', 'dport')
+
+  params = [
+    ('INPUT', 'in_interface', ports[0], UP_QUEUE),
+    ('OUTPUT', 'out_interface', ports[1], DOWN_QUEUE),
```

Comment:
It looks like "up" and "down" are not the correct terms anymore. Like, If you're running this on the client side, the common usage of "up" would mean "toward the server", but this is the opposite.

I think the solution is to change all instances of "up" and "down" to "incoming" and "outgoing." This will include some changes in the in web UI as well, but it shouldn't be anything too involved, just renaming.